### PR TITLE
feat: delay loading error-callsites module

### PR DESCRIPTION
**Pull request by astorm**
Closes: #2833

Delays loading of `error-callsites` to agent start to avoid unwanted `prepareStackTrace` side effects prior to the agent starting. 

### Checklist

- [x] Implement code
- [x] Add tests
- [x] Update TypeScript typings
- [x] ~Update documentation~
- [x] Add CHANGELOG.asciidoc entry

### CODE REVIEW

**Comment by astorm** [comment]
@trentm still working up a test or two on this one but wanted to run the idea of wrapping `error-callsites` in a lazy load callback and then explicitly loading it in `Agent.start`.  Was this what you had in mind or was there another approach you were considering?

**Review by trentm — CHANGES_REQUESTED** [review]

File: test/stacktraces/stacktraces.test.js [hunk 1]
```diff
@@ -398,5 +398,27 @@ tape.test('gatherStackTrace()', function (suite) {
     })
   })
 
+  tape.test('delayed error-callsite loading', function (t) {
```
[the remark this answers is not in the document]
Comment (trentm) [reply test/stacktraces/stacktraces.test.js:401]: Oh, I missed your comment about tests still being draft: https://github.com/elastic/apm-agent-nodejs/pull/2906#issuecomment-1236417093

File: lib/agent.js [hunk 2]
```diff
@@ -217,6 +217,9 @@ Agent.prototype.start = function (opts) {
   if (this.isStarted()) {
     throw new Error('Do not call .start() more than once')
   }
+  // load error-callsites on start
+  // https://github.com/elastic/apm-agent-nodejs/issues/2833
+  require('error-callsites')
```
[the remark this answers is not in the document]
Comment (trentm) [reply lib/agent.js:222]: I think I mostly answered the questions in your comment https://github.com/elastic/apm-agent-nodejs/pull/2906#issuecomment-1236417093 here.

[the remark this answers is not in the document]
Comment (astorm) [reply lib/agent.js:222]: Yeah, still writing the tests.  Moving the loading makes sense and an `initStackTraceCollection` function seems like an equally valid way to handle this so I'll do that -- just to say it out loud the trade-off there is `gatherStackTrace` will need a guard that bails early if stack trace collection hasn't been initialized.

[the remark this answers is not in the document]
Comment (trentm) [reply lib/agent.js:222]: > just to say it out loud the trade-off there is gatherStackTrace will need a guard that bails early if stack trace collection hasn't been initialized.

Yup, +1.

File: CHANGELOG.asciidoc [hunk 3]
```diff
@@ -60,6 +60,11 @@ paths are considered for body capture. ({pull}2873[#2873])
   For example `require('node:http')` will now be instrumented.
   ({issues}2816[#2816])
 
+- Agent will delay loading of the the `error-callsites` module until agent start time,
+  and will not load the module if the agent is disabled/un-active. This prevents the
+  setting of an Error.prepareCallStack handler and any other side-effects from 
+  `error-callsites` ({issues}2833[#2833] {pull}2906[#2906])
```
Comment (trentm) [inline CHANGELOG.asciidoc:66]:
```suggestion
- Agent will delay loading of the `error-callsites` module until agent start time,
  and will not load the module if the agent is disabled/inactive. This prevents the
  setting of an `Error.prepareStackTrace` handler until necessary for stacktrace
  collection. ({issues}2833[#2833] {pull}2906[#2906])
```

**Review by trentm — CHANGES_REQUESTED** [review]
Looks good.

**Review by trentm — APPROVED** [review]

**Pull request merged**